src/loc.ml

```diff
@@ -85,6 +101,14 @@ let print ppf loc =
           Format.fprintf pp "%s\n%*s\n" line
             stop_c
             (String.make len '^')
+        else
+          let rec num_of_digits n =
+            if n = 0 then 0  else 1 + (num_of_digits (n / 10))
+          in
+          let lines = file_lines path start.pos_lnum stop.pos_lnum in
+          List.iteri ~f:(fun i l ->
+            Format.fprintf pp "%*d: %s\n" (num_of_digits stop.pos_lnum) (start.pos_lnum + i) l)
```

Comment:
I believe that you can simplify that a bit by having `file_lines` return the line number as a string. You can even use this to compute the field width using `String.length (snd (List.last lines))` or something.

If this makes things more complicated, I'm fine with the current solution as well.

Reply:
Good point. I'll update with `file_lines` returning the line numbers as well